Add documentation for Mattermost Helm chart (#13728)
gravitational/teleport: docs/pages/access-controls/access-request-plugins/ssh-approval-mattermost.mdx

plugin to use these credentials and post messages in the right channels for your
workspace.

<Tabs dropdownCaption="Environment type" dropdownSelected="Executable">

> <TabItem label="Teleport Enterprise" options="Executable">

Author: Yeah, true

plugin to use these credentials and post messages in the right channels for your
workspace.

<Tabs dropdownCaption="Environment type" dropdownSelected="Executable">

> <TabItem label="Command line" options="Executable">

Reviewer: ```suggestion
<TabItem label="Command Line" options="Executable">
```

`identity` field with the path to the identity file you exported earlier and
comment out the other fields.

</TabItem>
<TabItem label="Teleport Enterprise " scope={["oss", "enterprise"]} options="Helm Chart">

**`address`**: Include the hostname and HTTPS port of your Teleport Cloud tenant
(e.g., `mytenant.teleport.sh:443`).

**`identitySecretName`**: Fill in the `identitySecretName` field with the name
of the Kubernetes secret you created earlier.

> If you exported an identity file earlier, fill in the `identity` field with the

Reviewer: Just double-checking that this sentence is accurate for the Helm chart:

> If you exported an identity file earlier, fill in the identity field with the path to the file and comment out the other fields.

Author: Oh, forgot to delete that sentence

path to the file and comment out the other fields.

</TabItem>
<TabItem label="Teleport Cloud " scope="cloud" options="Helm Chart">

**`address`**: Include the hostname and HTTPS port of your Teleport Cloud tenant
(e.g., `mytenant.teleport.sh:443`).

**`identitySecretName`**: Fill in the `identitySecretName` field with the name
of the Kubernetes secret you created earlier.

</TabItem>
</Tabs>

### `[mattermost]`

> <Tabs dropdownCaption="Environment type" dropdownSelected="Executable">

Reviewer: I think we can use a regular `Tabs` component here, without a dropdown menu, since there is one tab for a Helm chart and one tab for an executable

Author: Yeah, I was thinking about keeping the different axes on their "original" place, creating some consistency throughout the document, but let's try the other way. I'll change it.

`identity` field with the path to the identity file you exported earlier and
comment out the other fields.

</TabItem>
<TabItem label="Teleport Enterprise " scope={["oss", "enterprise"]} options="Helm Chart">

> **`address`**: Include the hostname and HTTPS port of your Teleport Cloud tenant

Reviewer: ```suggestion
**`address`**: Include the hostname and HTTPS port of your Teleport Proxy Service
```

> (e.g., `mytenant.teleport.sh:443`).

Reviewer: ```suggestion
(e.g., `teleport.example.com:443`).
```

severity = "INFO" # Logger severity. Could be "INFO", "ERROR", "DEBUG" or "WARN".
```
</TabItem>
<TabItem scopes={["cloud"]} label="Connect to the Proxy Service " options="Helm Chart">
```yaml
(!examples/resources/plugins/teleport-mattermost-helm-cloud.yaml!)
```
</TabItem>
<TabItem scopes={["oss", "enterprise"]} label="Connect to the Auth Service " options="Helm Chart">
```yaml
(!examples/resources/plugins/teleport-mattermost-helm-self.yaml!)
```
</TabItem>
</Tabs>

## Step 7/8. Test your Mattermost bot

> <Tabs dropdownCaption="Environment type" dropdownSelected="Executable">

Reviewer: Since there are only two options here, I think we can use a regular `Tab`s component here and include labels for "Executable" and "Helm Chart", rather than dropdown options.

DEBU   Watcher connected mattermost/main.go:260
DEBU   Mattermost API health check finished ok mattermost/main.go:19
```
</TabItem>
<TabItem label="Helm Chart" scope={["oss", "enterprise"]}>
After modifying your configuration, run the bot with the following command:

```code
$ helm upgrade --install teleport-plugin-mattermost teleport/teleport-plugin-mattermost --values teleport-mattermost-helm.yaml
```

To inspect the plugin's logs, use the following command:

```code
$ kubectl logs deploy/teleport-plugin-mattermost
```

Debug logs can be enabled by setting `log.severity` to `DEBUG` in

> `teleport-mattermost-helm.yaml` and execute the `helm upgrade ...` command

Reviewer: ```suggestion
`teleport-mattermost-helm.yaml` and executing the `helm upgrade ...` command
```
